vscode-extensions.ms-vscode.PowerShell: init at 2022.7.2

rhoriguchi opened the pull request:
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

rhoriguchi committed:
vscode-extensions.ms-vscode.PowerShell: init at 2022.7.2

superherointj commented:
@rhoriguchi Can you complete meta information?
Add download page as marketplace page.
Also, consider adding yourself as maintainer.

rhoriguchi commented:
> @rhoriguchi Can you complete meta information? Add download page as marketplace page (https://marketplace.visualstudio.com/items?itemName=ms-vscode.PowerShell). Also, consider adding yourself as maintainer.

As requested I've updated the metadata and added myself as a maintainer.

superherointj commented:
@rhoriguchi The package is erroring as:
`error: undefined variable 'maintainers'`
Likely, you have forgotten `with lib;` in meta.
Certify to build the package after every change.

superherointj commented on pkgs/applications/editors/vscode/extensions/default.nix:
```diff
@@ -1674,6 +1674,22 @@ let
+          version = "2022.6.3";
+          sha256 = "sha256-P8kakmpT3yAkKFroKttiLOV7PneOIoQOEtwImPIDHQY=";
+        };
+        meta = {
```
```suggestion
        meta = with lib; {
```

superherointj commented on pkgs/applications/editors/vscode/extensions/default.nix:
```diff
@@ -1674,6 +1674,22 @@ let
+          description = "PowerShell Language Support for Visual Studio Code.";
+          downloadPage = "https://marketplace.visualstudio.com/items?itemName=ms-vscode.PowerShell";
+          homepage = "https://github.com/PowerShell/vscode-powershell";
+          license = lib.licenses.mit;
```
```suggestion
          license = licenses.mit;
```

superherointj commented on pkgs/applications/editors/vscode/extensions/default.nix:
```diff
@@ -1674,6 +1674,22 @@ let
+          sha256 = "sha256-P8kakmpT3yAkKFroKttiLOV7PneOIoQOEtwImPIDHQY=";
+        };
+        meta = {
+          description = "PowerShell Language Support for Visual Studio Code.";
```
```suggestion
          description = "PowerShell Language Support for Visual Studio Code";
```

superherointj replied:
Please remove "." from description to comply with guidelines:

https://nixos.org/manual/nixpkgs/stable/#var-meta-description

superherointj replied:
Description could be improved too:
`A Visual Studio Code extension for PowerShell language support` (feel free to improve it further)

rhoriguchi commented:
@superherointj sorry for the hassle and thx for the reviews.

superherointj commented:
Result of `nixpkgs-review pr 182817` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vscode-extensions.ms-vscode.PowerShell</li>
  </ul>
</details>

superherointj commented on pkgs/applications/editors/vscode/extensions/default.nix:
```diff
@@ -1674,6 +1674,22 @@ let
+        mktplcRef = {
+          name = "PowerShell";
+          publisher = "ms-vscode";
+          version = "2022.6.3";
```
```suggestion
          version = "2022.7.2";
```

superherointj commented:
@rhoriguchi A new version is out. Can you update?

rhoriguchi commented:
> @rhoriguchi A new version is out. Can you update?

Is updated :)

superherointj commented:
Result of `nixpkgs-review pr 182817` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vscode-extensions.ms-vscode.PowerShell</li>
  </ul>
</details>

superherointj reviewed (commented):
LGTM
